### DISCUSSION TITLE
[Seeking Contributors] Port TonalJS to C++

### DISCUSSION THREAD

**Discussion by heyaphra** [tonaljs/tonal#353] [Ideas]
I'll keep it short:

No music theory library like this currently exists in any other programming language. I've tried alternatives and they do not get the job done. It would be of immense value to be able to use it in creative coding environments such as openFrameworks.

I've started this endeavor here: https://github.com/spidercatnat/tonal-cpp/tree/development

I am a C++ n00b and would love contributors.

That is all, thank you!

**Comment by nathanleiby** [comment]
Hi @spidercatnat! I'm also a C++ noob. If I were going about this, I might explore whether any existing tools would allow you to leverage this TonalJS codebase directly, rather than doing a full rewrite in C++. Here's a StackOverflow post suggesting many possible approaches: https://stackoverflow.com/questions/5192223/convert-javascript-code-to-c-code
